sycl/include/CL/sycl/ONEAPI/group_algorithm.hpp

```diff
@@ -190,6 +190,23 @@ using EnableIfIsNonNativeOp = cl::sycl::detail::enable_if_t<
         !cl::sycl::detail::is_native_op<T, BinaryOperation>::value,
     T>;
 
+template <typename T> struct is_native_function_object : std::false_type {};
+
+template <template <typename> class BinaryOperation, typename T>
+struct is_native_function_object<BinaryOperation<T>> {
+  static const bool value =
+      (sycl::detail::is_scalar_arithmetic<T>::value ||
+       sycl::detail::is_vector_arithmetic<T>::value ||
+       std::is_same<T, void>::value) &&
+      sycl::detail::is_native_op<T, BinaryOperation<T>>::value;
+};
+
+#if __cplusplus >= 201703L
+template <typename T>
```

Comment:
I think we may emulate `inline` variables even before c++17. The question is should we? If the answer is "Yes, we should" you may ask me how to do that. I suggested that way for one of the compiler features in the past.

To be clear, I am ok to leave it as is. Just wanted to point out that if it's important to have that API prior c++17 there is the way

Reply:
I'm leaning towards leaving it as it is.  The `_v` traits in the standard library (e.g. `std::is_execution_policy_v`) are only enabled for C++17 onwards, and I don't expect people to mix C++17-style traits with earlier standards.

My opinion would be different if there was a precedent for emulating standard SYCL `_v` traits in our implementation.  But I don't think there are any other standard traits implemented yet!